### PR TITLE
Feat/met 6069 Collect historical data for statistics-dashboard

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: metis-statistics-data/src/main/resources/application.properties.example
Comment (jeortizquan) [inline metis-statistics-data/src/main/resources/application.properties.example:1]: log4j2.xml file is missing. my suggestion would be to add it.
```
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="WARN">
  <Appenders>
    <Console name="Console" target="SYSTEM_OUT">
      <PatternLayout pattern="%d{yyyy-MM-dd'T'HH:mm:ss.SSSXXX} %-5p %C:%L [%t] - %m%n"/>
    </Console>
  </Appenders>
  <Loggers>
    <Root level="INFO">
      <AppenderRef ref="Console"/>
    </Root>
  </Loggers>
</Configuration>
```

File: metis-statistics-data/src/main/java/eu/europeana/metis/config/MongoSDDao.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+   * @param mongoClient       The mongo client.
+   * @param mongoDatabaseName The name of the database in the Mongo.
+   * @param createIndexes     The flag that initiates the database indexes
+   */
+  public MongoSDDao(MongoClient mongoClient, String mongoDatabaseName, boolean createIndexes) {
+    final MapperOptions mapperOptions = MapperOptions.builder()
+        .collectionNaming(NamingStrategy.identity()).build();
+    this.datastore = Morphia.createDatastore(mongoClient, mongoDatabaseName, mapperOptions);
+    this.datastore.getMapper().map(Historical.class);
+    if(createIndexes){
```
Comment (jeortizquan) [inline metis-statistics-data/src/main/java/eu/europeana/metis/config/MongoSDDao.java]: Just check the formatting for this class and the rest of the project, the `if` and parenthesis are together.

File: metis-statistics-data/src/main/java/eu/europeana/metis/config/DataAccessConfigException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package eu.europeana.metis.config;
+
+public class DataAccessConfigException extends Exception {
+
+  public DataAccessConfigException(String message) {
+    super(message);
+  }
+
+  public DataAccessConfigException(String message, Throwable cause) {
```
Comment (jeortizquan) [inline metis-statistics-data/src/main/java/eu/europeana/metis/config/DataAccessConfigException.java]: remove. this constructor is not being used.

File: metis-statistics-data/src/main/java/eu/europeana/metis/HistoricalDataScript.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    public HistoricalDataScript(ConfigurationPropertiesHolder propertiesHolder) {
+        this.propertiesHolder = propertiesHolder;
+    }
+
+    @Override
+    public void run(String... args) throws DataAccessConfigException {
+
+        File folder = new File("metis-statistics-data/src/main/resources/historicalData");
+        File[] dataFiles = Objects.requireNonNull(folder.listFiles());
```
Comment (jeortizquan) [inline metis-statistics-data/src/main/java/eu/europeana/metis/HistoricalDataScript.java]: my suggestion is to add a friendly message to this
```
Objects.requireNonNull(folder.listFiles()) 
```
to for example:
```
Objects.requireNonNull(folder.listFiles(),"a list of files required by <add description>");
```

**Review by jeortizquan — COMMENTED** [review]
There are small suggestions, and check the `metis-framework` coding style. Seems is not activated on the project.

File: metis-statistics-data/src/main/resources/application.properties.example
Comment (JoanaCMS) [reply metis-statistics-data/src/main/resources/application.properties.example:1]: Added!

File: metis-statistics-data/src/main/java/eu/europeana/metis/config/MongoSDDao.java (hunk 1 above)
Comment (JoanaCMS) [reply metis-statistics-data/src/main/java/eu/europeana/metis/config/MongoSDDao.java]: Weird, the settings seems ok, but I'll fix this

File: metis-statistics-data/src/main/java/eu/europeana/metis/config/DataAccessConfigException.java (hunk 2 above)
Comment (JoanaCMS) [reply metis-statistics-data/src/main/java/eu/europeana/metis/config/DataAccessConfigException.java]: Removed

File: metis-statistics-data/src/main/java/eu/europeana/metis/HistoricalDataScript.java (hunk 3 above)
Comment (JoanaCMS) [reply metis-statistics-data/src/main/java/eu/europeana/metis/HistoricalDataScript.java]: Wrote a slightly different message, let me know what you think

Comment (jeortizquan) [reply metis-statistics-data/src/main/java/eu/europeana/metis/HistoricalDataScript.java]: is fine :smiley:

**Pull request merged**